Datastore/tests/Unit/DatastoreClientTest.php

```diff
@@ -749,4 +807,15 @@ private function mutationsProviderProvider($id, $partialKey = false)
             ['upsert', $mutation, clone $key, $id],
         ];
     }
+
+    private function compareResult($expected, $actual)
```

Comment:
This is repeated in `LookupTest` as well as `EntityMapperTest`, also the one in `LookupTest` is cleaner. 
Try moving them to a common place to avoid code duplication (looks difficult to achieve)
And simplify the other functions.

Reply:
I agree, I think the best solution is to extract the `compareResult` method properly to [Core/src/Testing](https://github.com/googleapis/google-cloud-php/tree/main/Core/src/Testing/) as this repository contains everthing related to testing.

Since this change would involve Datastore as well as Firestore, I will fix this in another PR.

Reply:
Meanwhile I've made the methods simpler in this PR